system-parachains/asset-hubs/asset-hub-kusama/src/verify_xcm_upgrade_migration.rs

```diff
@@ -0,0 +1,223 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod migration {
+	use crate::ForeignAssetsInstance;
+	#[cfg(feature = "try-runtime")]
+	use crate::{vec, Vec};
+	#[cfg(feature = "try-runtime")]
+	use codec::{Decode, Encode};
+	use frame_support::traits::OnRuntimeUpgrade;
+	#[cfg(feature = "try-runtime")]
+	use frame_support::weights::Weight;
+	#[cfg(feature = "try-runtime")]
+	use sp_core::Get;
+	#[cfg(feature = "try-runtime")]
+	use sp_runtime::TryRuntimeError;
+
+	#[cfg(feature = "try-runtime")]
+	const LOG_TARGET: &str = "runtime::verify_xcm_upgrade_migration";
+
+	mod asset_as_v4 {
+		use crate::ForeignAssetsInstance;
+		use frame_support::{storage_alias, Blake2_128Concat};
+
+		#[storage_alias]
+		pub type Asset<T: frame_system::Config + pallet_assets::Config<ForeignAssetsInstance>> =
+			StorageMap<
+				pallet_assets::Pallet<T, ForeignAssetsInstance>,
+				Blake2_128Concat,
+				xcm::v4::Location,
+				pallet_assets::AssetDetails<
+					<T as pallet_assets::Config<ForeignAssetsInstance>>::Balance,
+					<T as frame_system::Config>::AccountId,
+					pallet_assets::DepositBalanceOf<T, ForeignAssetsInstance>,
+				>,
+			>;
+
+		#[storage_alias]
+		pub type Pools<T: pallet_asset_conversion::Config> = StorageMap<
+			pallet_asset_conversion::Pallet<T>,
+			Blake2_128Concat,
+			<T as pallet_asset_conversion::Config>::PoolId,
+			pallet_asset_conversion::PoolInfo<<T as pallet_asset_conversion::Config>::AssetKind>,
+		>;
+	}
+
+	mod asset_as_v5 {
+		use crate::ForeignAssetsInstance;
+		use frame_support::{storage_alias, Blake2_128Concat};
+
+		#[storage_alias]
+		pub type Asset<T: frame_system::Config + pallet_assets::Config<ForeignAssetsInstance>> =
+			StorageMap<
+				pallet_assets::Pallet<T, ForeignAssetsInstance>,
+				Blake2_128Concat,
+				xcm::v4::Location,
+				pallet_assets::AssetDetails<
+					<T as pallet_assets::Config<ForeignAssetsInstance>>::Balance,
+					<T as frame_system::Config>::AccountId,
+					pallet_assets::DepositBalanceOf<T, ForeignAssetsInstance>,
+				>,
+			>;
+
+		#[storage_alias]
+		pub type Pools<T: pallet_asset_conversion::Config> = StorageMap<
```

Comment:
@claravanstaden this is exactly the same as for asset_as_v4, do we really need this?

Reply:
I should have specified the xcm version explicitly, done in https://github.com/polkadot-fellows/runtimes/pull/753/commits/1f88a47b7d3a6d16a4be5607fa4935cde144ed24.